### ISSUE TITLE
MPC [Video Renderer]-HC Crashes /w Error On Videos As Of Late

### ISSUE THREAD

**Issue by alienfxfiend** [clsid2/mpc-hc#1473]
I didn't have this problem before —it cropped up in a recent update. When I play a video file (MP3's not problem, video issue) using 'MPC Video Renderer', it crashes /w (may have got the L's\1's\i's mixed up due to my font) following error box (have to kill the process)::
"An error has occurred. MPC-HC will close now. Exception: ACCESS VIOLATION Crashing module: C:\WINDOWS\SYSTEM32\Igd101ymd64.dll Offset: 0x1A520, Codebase: 0x7FFF1F190000 The thread 1176 tried to read memory at address 0x19175000."
I switched to 'Enhanced Video Renderer' & then it works normally, though I prefer the 'MPC Video Renderer'. If this is a Nightly issue that will be fixed let me know: I use Windows 10 Enterprise Edition on a Dell 3543 laptop /w Intel HD Graphics (I spruced up the Graphics Card settings a bit long time back.) BTW I had manually switched all the Codecs to 'LAV Audio\ VIdeo' K-Lite Codec Pack's ones —but that was some months ago & didn't result in the recent crash. This is obviously a Graphics Card glitch, but I haven't updated any drivers or anything, would like a fix in the upcoming K-Lite Codec Pack. No new Windoes updates or anything —I can't explain it !

**Comment by clsid2** [comment]
Igd10iumd64.dll belongs to Intel graphics driver.

Try with "use Direct3d 11" unchecked in MPCVR settings.

Make sure that "Contrast Enhancement" is disabled in the driver settings. That feature only causes troubles.

**Comment by alienfxfiend** [comment]
> Igd10iumd64.dll belongs to Intel graphics driver.
> 
> Try with "use Direct3d 11" unchecked in MPCVR settings.
> 
> Make sure that "Contrast Enhancement" is disabled in the driver settings. That feature only causes troubles.

Yeah /w DirectX 11 unchecked in MPC-VR, it doesn't crash —that fixed it, thanks ! Yeah I heavily tweaked the contrast & color settings, I'll stop that if possible !

**Comment by clsid2** [comment]
That is an option separate from the regular color/contrast settings. But it is possible that the crash is related to custom driver settings. That could explain why only some people have the problem.

If you ever reset the driver settings, test with "use Direct3d 11" enabled again. I am curious if it still crashes then. Would be useful to know if a specific driver setting triggers the crash.

**Comment by alienfxfiend** [comment]
> That is an option separate from the regular color/contrast settings. But it is possible that the crash is related to custom driver settings. That could explain why only some people have the problem.
> 
> If you ever reset the driver settings, test with "use Direct3d 11" enabled again. I am curious if it still crashes then. Would be useful to know if a specific driver setting triggers the crash.

Alright I will test it now, & yeah I meant Color\ Contrast settings tweaked in the Intel HD Graphics Control Panel :] I increased gamma & the rest for readability.

**Comment by alienfxfiend** [comment]
Yeah it still crashes /w Display -> Color Settings & Video -> Color Enhancement set to defaults, so there must be a glitch in MPC-VR :[ I could try after a restart as I had HIPS + Auto-Containment enabled in Comodo Internet Security & it prompted for permissions. It used to work fine before, one of the newer problems that have cropped up (Nightly Builds maybe ?) ! Disabling DirectX 11 seems to fix it, my guess is that K-Lite Codec Pack is geared towards Win 11 now & I'm still using Win 10 indefinitely. Or it could be that I had enabled MPC-VR DX 11 accidentally recently. EVR doesn't crash —what does it use ?

**Comment by clsid2** [comment]
There is no evidence that this is because of a bug in MPCVR. The crash occurs in the Intel driver. And it also seems to be mostly happening on HD4600 and HD5500 GPU models. Which one do you have?

DirectX 11 has no relation at all with Windows 11! Those version numbers are unrelated. Your Windows 10 supports all versions of DirectX up to and including DX12!

The "use dx 11" option was disabled by default in the past. So that can be why you didn't have the issue before. It can be disabled again for the affected systems. But I prefer finding out why it happens, so a better solution can be found.

Did the crash happen with any video, including low resolution ones? If it only happened with specific ones, like 4K resolution or 10bit bitdepth, then that is something we can investigate further.

**Comment by alienfxfiend** [comment]
Intel HD Graphics "Driver Date: 16/10/2017" "Driver Version: 20.19.15.4835" (Device Manager) || Intel HD Graphics Control Panel: Microsoft DirectX: "Runtime Version: 12.0" "Hardware Supported Version: 11.1". I use the Dell website driver not Intel DCH driver (I have disabled Driver Updates via Windows Update (Group Policy) because the Intel DCH driver (which is supposed to support DX12) doesn't have a Control Panel GUI ! Never proved to be a problem before though. I have a Dell Inspiron 15 3000 Series 3543 /w Intel Celeron 3205U @ 1.5Ghz Dual-Core (laptop I bought it in 2012 or so.) I don't really know how to get the Graphics Card model number sorry. It's c00 I'll use it w/o DX11 ! I tested it /w MP4 + MKV & they seem to crash but not MP3's (videos not audio files.)
